src/nupic/algorithms/ColumnPooler.cpp

```diff
@@ -0,0 +1,484 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013, Numenta, Inc.
+ * Copyright (C) 2019, David McDougall
+ *
+ * Unless you have an agreement with Numenta, Inc., for a separate license for
+ * this software code, the following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ----------------------------------------------------------------------
+ */
+
+/** @file
+ * Implementation of ColumnPooler
+ */
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <iomanip> // std::setprecision
+#include <functional> // function
+
+// #include <nupic/algorithms/ColumnPooler.hpp>
+#include <nupic/types/Types.hpp>
+#include <nupic/types/Serializable.hpp>
+#include <nupic/types/Sdr.hpp>
+#include <nupic/utils/SdrMetrics.hpp>
+#include <nupic/algorithms/Connections.hpp>
+#include <nupic/math/Math.hpp>
+#include <nupic/math/Topology.hpp>
+
+using namespace std;
```

Comment:
make this a namespace nupic { algorithms { columns_pooler { 

to avoid including the "using ns" globally